Team9Project/src/com/indragie/cmput301as1/OnlineManager.java

```diff
@@ -0,0 +1,104 @@
+/* 
+ * Copyright (C) 2015 Jimmy Ho
+ * 
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *  
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.indragie.cmput301as1;
+
+import java.io.IOException;
+import java.util.LinkedList;
+
+import com.indragie.cmput301as1.ElasticSearchAPIClient.APICall;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.Response;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.util.Log;
+
+public class OnlineManager<T extends ElasticSearchDocument> extends BroadcastReceiver{
+
+	private LinkedList<ElasticSearchAPIClient.APICall<T>> stack = new LinkedList<ElasticSearchAPIClient.APICall<T>>(); 
+	
+	public void enqueueCall(ElasticSearchAPIClient.APICall<T> call) {
+		stack.add(call);
+		checkForNetwork();
+	}
+	
+	private Context context;
+
+	public void checkForNetwork() {
+		// if there is something inside of the stack, do this, else do nothing
+		if (stack.size() > 0) {	
+			
+			if (isNetworkAvailable(context) == true) {
+				// we are connected so then we want to go through with our stack calls
+				
+				// removes and returns the first item from the linked list
+				APICall<T> call = stack.getFirst();
+				// now do something with the call
+				call.enqueue(new ElasticSearchAPIClient.APICallback<T>() {
+					@Override
+					public void onFailure(Request request, Response response, IOException e) {
+						// on failure you want to check the network
+						if (isNetworkAvailable(context)==false) {
```

Comment:
Actually, this doesn't need to be there at all because there's no reason to check the status of the network if a request fails. If the request fails, it could either mean the network was down or that the network was up but the request didn't succeed. Basically, you need to check whether `response` is non-null. In that scenario, it means the request executed but the server returned a bad HTTP status code. We can't really do anything about that, so the API call should be removed from the stack and an error should be displayed to the user. If response is null, that means the request didn't execute at all. In that scenario, it is likely that there's something wrong with the network connection so we should wait until the connection is available to retry (do nothing here).